courses/views.py

```diff
@@ -20,13 +23,51 @@ class CourseViewSet(viewsets.ModelViewSet):
     lookup_field = 'uuid'
     serializer_class = CourseSerializer
 
+    def get_object(self):
+        """
+        Courses can be looked up via the uuid passed via kwargs (default
+        behavior) or through the course_id attribute in the payload.
+        """
+        queryset = self.get_queryset()
+
+        try:
+            obj = super(CourseViewSet, self).get_object()
+        except (AssertionError, Http404):
+            try:
+                obj = queryset.get(course_id=self.request.data.get('course_id'))
+            except Course.DoesNotExist:
+                pass
+        if not obj:
```

Comment:
obj not being defined isn't actually a code path we can hit. I've left a comment to this effect.